### PR TITLE
Update Core ML Backend Doc

### PR DESCRIPTION
**Pull request by YifanShenSZ**
Update Core ML backend doc on:
1. Partitioner
2. Quantizer

### CODE REVIEW

File: backends/apple/coreml/README.md [hunk 1]
```diff
@@ -20,41 +22,104 @@ Core ML is an optimized framework for running machine learning models on Apple d
...
+source_model = Model()
+example_inputs = (torch.randn((1, 3, 256, 256)), )
+
+pre_autograd_aten_dialect = capture_pre_autograd_graph(model, example_inputs)
+
+quantization_config = LinearQuantizerConfig.from_dict(
+    {
+        "global_config": {
+            "quantization_scheme": QuantizationScheme.symmetric,
+            "milestones": [0, 0, 10, 10],
```
Comment (pulkital) [inline backends/apple/coreml/README.md:105]: Could you remove milestones from here? It is actually not being used by the `CoreMLQuantizer` and would be confusing for users. It is used by `LinearQuantizer` in `ct.optimize.torch`.

Comment (YifanShenSZ) [reply backends/apple/coreml/README.md]: Sure, removed as suggested

**Comment by cymbalrush** [comment]
Rebased on main.

File: examples/apple/coreml/README.md [hunk 2]
```diff
@@ -27,40 +27,50 @@ cd executorch
...
 add_mul
+dl3
+edsr
+emformer_join
+emformer_predict
+emformer_transcribe
+ic3
 ic4
 linear
+llama2
```
Comment (cccclai) [inline examples/apple/coreml/README.md:68]: have you tried llama3? If llama2 works, llama3 should be just out of box.

**Review by cccclai — APPROVED** [review]

**Review by shoumikhin — APPROVED** [review]

File: examples/apple/coreml/README.md (hunk 2 above)
Comment (cymbalrush) [reply examples/apple/coreml/README.md:68]: Not yet, going to try today.

**Pull request closed without merging**